lib/testing/FakeVisPattern.hpp

```diff
@@ -0,0 +1,225 @@
+/*****************************************
+@file
+@brief Patterns for generating fake visibility data
+- FakeVisPattern
+- DefaultVisPattern
+- FillIJVisPattern
+- FillIJMissingVisPattern
+- PhaseIJVisPattern
+- ChimeVisPattern
+- TestPatternSimpleVisPattern
+- TestPatternFreqVisPattern
+- TestPatternInputVisPattern
+*****************************************/
+#ifndef FAKE_VIS_PATTERN_HPP
+#define FAKE_VIS_PATTERN_HPP
+
+#include "Config.hpp"
+#include "factory.hpp"
+#include "kotekanLogging.hpp"
+#include "visBuffer.hpp"
+
+#include <stdint.h>
+#include <string>
+
+/**
+ * @class FakeVisPattern
+ * @brief Base class for fakeGpu test patterns.
+ *
+ * @conf  num_elements          Int. The number of elements (i.e. inputs) in
+ *                              the correlator data
+ * @conf  block_size            Int. The block size of the packed data
+ * @conf  samples_per_data_set  Int. FPGA seq ticks per frame. Only use for
+ *                              `pre_accumulate`.
+ * @conf  num_freq_in_frame     Int. Number of frequencies packed into each GPU
+ *                              frame.
+ **/
+class FakeVisPattern : public kotekan::kotekanLogging {
+public:
+    /**
+     * @brief Create a class that makes a fakeGpu test mode.
+     *
+     * @param  config  Kotekan configuration.
+     * @param  path    Path to start search from. Will generally be the path for
+     *                 the process creating the pattern.
+     **/
+    FakeVisPattern(kotekan::Config& config, const std::string& path);
+
+    virtual ~FakeVisPattern() = default;
+
+    /**
+     * @brief Fill the data with a test pattern.
+     *
+     * @note You can also modify the metadata (for example to set the number of
+     *       lost frames), but be careful doing this if there are multiple
+     *       frequencies in the buffer.
+     *
+     * @param  data       Data for the specific frequency in the buffer.
```

Comment:
CI failed because of the parameters listed here incorrectly.

Reply:
Ooops I've pushed a fix. Should be building now.